### PR TITLE
orcid: push user literature when setting enabled

### PR DESCRIPTION
**Pull request by harunurhan**
* INSPIR-3171

### CODE REVIEW

File: backend/inspirehep/orcid/tasks.py [hunk 1]
```diff
@@ -268,3 +269,13 @@ def _find_user_matching(orcid, email):
     if user_identity:
         return user_identity.user
     return User.query.filter_by(email=email).one_or_none()
+
+
+@shared_task(ignore_result=False, max_retries=3)
```
Comment (drjova) [inline backend/inspirehep/orcid/tasks.py]: with ``ignore_results=False`` the endpoint will wait to schedule all the tasks first

File: backend/inspirehep/accounts/views.py [hunk 2]
```diff
@@ -88,4 +90,17 @@ def set_orcid_push_setting():
...
+        tokens = orcid_account.remote_tokens
+
+        if len(tokens) != 1:
+            raise ValueError(
+                f"One token per remote account is expected, but found {len(tokens)} for {orcid}"
+            )
+
+        push_account_literature_to_orcid.apply_async(
+            kwargs={"orcid": orcid, "token": tokens[0].access_token}
+        )
```
Comment (drjova) [inline backend/inspirehep/accounts/views.py]: Actually I'd just get ``[0]`` and raise on IndexError instead of checking how many tokens we have.

Comment (harunurhan) [reply backend/inspirehep/accounts/views.py]: whole thing (and many more things) on relies on the fact that we have a single token per account, and all accounts are `orcid`, that's why there is the check.

File: backend/inspirehep/orcid/tasks.py (hunk 1 above)
Comment (harunurhan) [reply backend/inspirehep/orcid/tasks.py]: yeah copy-paste, needs `True` indeed.

File: backend/inspirehep/orcid/tasks.py [hunk 3]
```diff
@@ -268,3 +269,13 @@ def _find_user_matching(orcid, email):
     if user_identity:
         return user_identity.user
     return User.query.filter_by(email=email).one_or_none()
+
+
+@shared_task(max_retries=3)
```
Comment (drjova) [inline backend/inspirehep/orcid/tasks.py]: ``max_retries`` alone I think will do nothing you have to ``self.retry`` with ``bind=True`` or you have to specify ``autoretry_for`` and specify the exception (i.e. can be something silly like ``autoretry_for=(Exception,)``) do we really want this?

Comment (harunurhan) [reply backend/inspirehep/orcid/tasks.py]: yeah this one is from a copy-paste, tbh I don't think we need retries for this

File: backend/tests/helpers/factories/models/user_access_token.py [hunk 4]
```diff
@@ -48,12 +49,14 @@ def _create(
...
-            remote_account = RemoteAccount(
+            remote_token = RemoteToken.create(
                 user_id=user.get_id(),
                 client_id="orcid",
+                token=token,
+                secret=None,
                 extra_data={"orcid": orcid, "allow_push": allow_push},
             )
-            db.session.add(remote_account)
+            db.session.add(remote_token)
```
Comment (drjova) [inline backend/tests/helpers/factories/models/user_access_token.py]: Does it already https://github.com/inveniosoftware/invenio-oauthclient/blob/master/invenio_oauthclient/models.py#L240

**Pull request merged**